src/window/User Guide.md

```diff
@@ -0,0 +1,132 @@
+
+Because of this abstraction, some of the details of the piston API may seem unnecessary or useless. The purpose of this guide is to give concrete guidelines on the creation and usage of piston windows.
+
+If you mostly know what you're doing, but need a quick refresher, see the quick reference at the bottom of this guide.
+
+# Windows
+
+## Choosing a Backend
+
+There are a few available backends for piston, and it's important to choose one for your project.
+For the start of most projects, any backend will work. If you're coming from another programming language like C or C++, you should use a backend that you've worked with in the past. Beyond these superficial differences, there are other things to consider:
+- A backend may look different on other Operating Systems.
+- A backend may provide extra functionality, such as better touch support, or better controller support.
+- A backend may perform better for the specific things your code does.
+
+In any event, it's very easy to switch backends at any time.
+
+## Using your Backend
+
+Each backend has its own structure that implements `Window`. They're uniquely named, usually `<lib name>Window`. It's convention to rename this main window structure to `Window` for clarity. It also makes switching backends much easier, because only the use statement needs to be changed.
+
+Example of the use statement, with the Sdl2 backend:
+
+```Rust
+extern crate sdl2_window;
+
+use sdl2_window::Sdl2Window as Window;
+```
+
+The rest of this guide will follow this convention, and do not confuse the type `Window` with the trait `window::Window`
+
+## Creating Windows
+
+There are two ways to create a window: using `Window::new()`, or `WindowSettings::new().build()` While both valid, the convention is to use `WindowSettings` to make a window.
+
+### Using `WindowSettings::new().build()`
+
+This is the normal way of creating a new window. Generally, the window's settings are set up using a method chain, like this:
+
+```Rust
+// Note: The type of main_window must be specified; .build() can return many different
+//	       types.
+let mut main_window: Window = WindowSettings::new()
+										.exit_on_esc(false)
+										.fullscreen(false)
+										.size(500, 500)
+										.build().unwrap();
+```
+
+The build method eventually calls `Window::new()`, but lets backends execute code before creating a window. It also allows backends to return an error. It is rare for an error to occur, though, so normally an `.unwrap()` is sufficient.
+
+It is also useful to note that `WindowSettings` objects remain valid after `build()`, and can be used to make more than one window.
+
+Here is an example of using a `WindowSettings` object twice:
+
+```Rust
+let window_settings = WindowSettings::new().size(500,500);
+
+let mut first_window: Window = window_settings.build().unwrap();
+
+let mut second_window: Window = window_settings.build().unwrap();
```

Comment:
Added an example for using multiple windows https://github.com/PistonDevelopers/piston-examples/pull/351. Currently you have to dig down to the `glutin::Window` object to hide a window properly.